private/buf/bufwork/workspace_builder.go

```diff
@@ -23,29 +23,38 @@ import (
 	"github.com/bufbuild/buf/private/buf/bufconfig"
 	"github.com/bufbuild/buf/private/bufpkg/bufmodule"
 	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmodulebuild"
-	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
 	"github.com/bufbuild/buf/private/pkg/normalpath"
 	"github.com/bufbuild/buf/private/pkg/storage"
 )
 
-type workspace struct {
-	// bufmoduleref.ModuleIdentity -> bufmodule.Module
-	namedModules map[string]bufmodule.Module
-	allModules   []bufmodule.Module
+type workspaceBuilder struct {
+	configProvider      bufconfig.Provider
+	moduleBucketBuilder bufmodulebuild.ModuleBucketBuilder
+	moduleCache         map[string]*cachedModule
```

Comment:
The `WorkspaceBuilder` seems to live around across invocations of `bufwire` functions, which means this cache is long-lived. Given that we never clear this cache, in a service context, this could eventually result in running out of memory. Two options:

1. Make this a LRU cache.
2. (probably more preferred) make `workspaceBuilder` something constructed for a single `GetImageConfig/GetModuleConfig` call.